test/functional/mempool_limit.py

```diff
@@ -50,5 +50,18 @@ def run_test(self):
         assert_equal(self.nodes[0].getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
         assert_greater_than(self.nodes[0].getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
 
+        self.log.info('Create a mempool tx that will not pass mempoolminfee')
+        us0 = utxos.pop()
+        inputs = [{ "txid" : us0["txid"], "vout" : us0["vout"]}]
+        outputs = {self.nodes[0].getnewaddress() : 0.0001}
+        tx = self.nodes[0].createrawtransaction(inputs, outputs)
+        self.nodes[0].settxfee(relayfee) # specifically fund this tx with a fee < mempoolminfee, >= than minrelaytxfee
+        txF = self.nodes[0].fundrawtransaction(tx)
+        self.nodes[0].settxfee(0) # return to automatic fee selection
+        tx_hex = self.nodes[0].signrawtransaction(txF['hex'])['hex']
+        tx_id = self.nodes[0].decoderawtransaction(tx_hex)["txid"]
+        assert_raises_rpc_error(-26, "66: mempool min fee not met", self.nodes[0].sendrawtransaction, tx_hex)
+        assert(tx_id not in self.nodes[0].getrawmempool())
```

Comment:
I guess there is no need to assert this, `sendrawtransaction` above failed. This `assert` makes sense when testing `sendrawtransaction` failure, so move it there if it's missing.

Reply:
Fair enough, removed that and generally simplified the test.

Fyi I was following the example set in: https://github.com/bitcoin/bitcoin/blob/master/test/functional/mining_prioritisetransaction.py#L124